openassessment/workflow/models.py

```diff
@@ -239,7 +240,8 @@ def get_score(self, assessment_requirements, step_for_name):
                     else:
                         requirements = assessment_requirements.get(assessment_step_name, {})
                     score = get_score_func(self.submission_uuid, requirements)
-                    break
+                    if score:
+                        break
```

Comment:
I don't mean checking to see if a score is available, but whether you should wait for one module to return a score instead of moving onto the next one. 

So you can test the following scenario:
* Create an ORA2 problem with Peer Assessment, Self Assessment
* Go in as a student, submit an essay
* Grade some peers
* Complete your self assessment
At this point, you should be waiting for grades from your peers, and you should not have a score. I believe with your change, you will have a score, based on your self assessment.